testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeCheckConstraintCompatibility.java

```diff
@@ -159,4 +159,40 @@ public void testMetadataOperationsRetainCheckConstraint()
             dropDeltaTableWithRetry("default." + tableName);
         }
     }
+
+    @Test(groups = {DELTA_LAKE_OSS, DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testUnsupportedCheckConstraintExpression()
+    {
+        String tableName = "test_unsupported_check_constraints_" + randomNameSuffix();
+        onDelta().executeQuery("CREATE TABLE default." + tableName + " (a INT, b INT) " +
+                "USING DELTA " +
+                "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'");
+
+        try {
+            // This constraint should be changed to a new one if the connector supports the expression
+            onDelta().executeQuery("ALTER TABLE default." + tableName + " ADD CONSTRAINT test_constraint CHECK (a = abs(b))");
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1, -1)");
+
+            assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (1, -1)"))
+                    .hasMessageContaining("Unsupported Spark expression");
```

Comment:
It's included in the error message. I intentionally avoid asserting the expression here because it's unrelated to the test purpose and we may change the expression in the future.